### PR TITLE
Adding linting

### PR DESCRIPTION
**Pull request by hbaqai**
Added eslint and started applying linting rules.

### CODE REVIEW

**Comment by hbaqai** [comment]
The test that is currently failing started to do so before these changes were added. The failure is that the sdk does not timeout. This works fine against production servers, but not with our http mockup library (Nock). We have filed an issue with Nock: https://github.com/node-nock/nock/issues/754

File: spec/opentok_spec.js [hunk 1]
```diff
@@ -554,37 +492,31 @@ describe('Archiving', function () {
     it('should return an error if archive ID is invalid', function (done) {
       nock('https://api.opentok.com:443')
         .delete('/v2/partner/APIKEY/archive/AN-INVALID-ARCHIVE-ID')
-        .reply(404, '{ "message" : "Not found. You passed in an invalid archive ID." }',
-        {
-          server: 'nginx',
+        .reply(404, '{ "message" : "Not found. You passed in an invalid archive ID." }', { server: 'nginx',
```
Comment (davemun) [inline spec/opentok_spec.js]: This whitespacing doesn't match the other nocks

File: spec/opentok_spec.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ describe('Archiving', function () {
     it('should return an Archive', function (done) {
       nock('https://api.opentok.com:443')
         .post('/v2/partner/APIKEY/archive', { sessionId: '1_MX4xMDB-MTI3LjAuMC4xflR1ZSBKYW4gMjggMTU6NDg6NDAgUFNUIDIwMTR-MC43NjAyOTYyfg', name: 'Bob' })
-        .reply(200, '{\n  "createdAt" : 1391149936527,\n  "duration" : 0,\n  "id" : "4072fe0f-d499-4f2f-8237-64f5a9d936f5",\n  "name" : "Bob",\n  "partnerId" : "APIKEY",\n  "reason" : ",\n  "sessionId" : "1_MX4xMDB-MTI3LjAuMC4xflR1ZSBKYW4gMjggMTU6NDg6NDAgUFNUIDIwMTR-MC43NjAyOTYyfg",\n  "size" : 0,\n  "status" : "started",\n  "url" : null\n}',
+        .reply(200, '{\n  "createdAt" : 1391149936527,\n  "duration" : 0,\n  "id" : "4072fe0f-d499-4f2f-8237-64f5a9d936f5",\n  "name" : "Bob",\n  "partnerId" : "APIKEY",\n  "reason" : "",\n  "sessionId" : "1_MX4xMDB-MTI3LjAuMC4xflR1ZSBKYW4gMjggMTU6NDg6NDAgUFNUIDIwMTR-MC43NjAyOTYyfg",\n  "size" : 0,\n  "status" : "started",\n  "url" : null\n}',
```
Comment (davemun) [inline spec/opentok_spec.js]: This .reply() third argument whitespacing is inconsistent with the rest of the file such as L#40 and onwards - should rectify to match.

**Comment by davemun** [comment]
+1 pending whitespacing comment

**Comment by davemun** [comment]
Accidentally clicked close, oops.

**Comment by hbaqai** [comment]
Addressed @davemun comment and merging after verbal +1 from him

**Pull request merged**